### PR TITLE
CASMPET-6582 Add roll spire postgres

### PR DESCRIPTION
**Pull request by kburns-hpe**
# CASMPET-6582
spire-postgres fails restart after the spire chart has been upgraded. This causes the postgres operator to mark it as 'SyncFailed'. Rolling the statefulset here should fix the issue.

NOTE: This section uses kubectl rollout status to wait for the pod restarts to complete before continuing. This only works on a node with a RollingUpdate strategy. After these nodes are rolled the operator will have changed the strategy to OnDelete. This command will fail after this happens.

# CASMINST-6424
Move PREFLIGHT_CHECK to bottom

Moved This check should be run last in order to validate that all the previoussteps are considered.

### CODE REVIEW

File: upgrade/scripts/upgrade/prerequisites.sh [hunk 1]
```diff
@@ -520,6 +520,34 @@ else
...
+if [[ ${state_recorded} == "0" ]]; then
+    echo "====> ${state_name} ..." | tee -a "${LOG_FILE}"
+  kubectl rollout restart -n spire statefulset spire-postgres
+  echo "Waiting for spire-postgres rollout to be completed"
+  kubectl rollout status -n spire statefulset spire-postgres
+  echo "Waiting for postgres operator to consider spire-postgres healthy"
+      loop_idx=0
+      healthy="false"
+      # wait up to 2 minutes for spire-postgres to be marked as healthy
+      while [[ ${loop_idx} -lt 18 && ${in_sync} == "no" ]]; do
```
Comment (bo-quan) [inline upgrade/scripts/upgrade/prerequisites.sh:534]:
```suggestion
      while [[ ${loop_idx} -lt 24 && ${healthy} == "false" ]]; do
```

**Review by bo-quan — CHANGES_REQUESTED** [review]

**Review by bo-quan — APPROVED** [review]

File: upgrade/scripts/upgrade/prerequisites.sh [hunk 2]
```diff
@@ -520,6 +520,34 @@ else
     echo "====> ${state_name} has been completed" | tee -a "${LOG_FILE}"
 fi
 
+state_name="ROLL_SPIRE_POSTGRES"
+state_recorded=$(is_state_recorded "${state_name}" "$(hostname)")
+if [[ ${state_recorded} == "0" ]]; then
+    echo "====> ${state_name} ..." | tee -a "${LOG_FILE}"
```
Comment (mitchty) [inline upgrade/scripts/upgrade/prerequisites.sh:526]: nit, all the code here is inconsistently indented by 2/4 spaces e.g the echo's don't line up with the general indentation consistently.

**Review by mitchty — APPROVED** [review]

File: upgrade/scripts/upgrade/prerequisites.sh (hunk 2 above)
Comment (kburns-hpe) [reply upgrade/scripts/upgrade/prerequisites.sh:526]: I had to turn off my autoformat for this file. shfmt changes a ton of stuff. I'll fix formatting.

Comment (kburns-hpe) [reply upgrade/scripts/upgrade/prerequisites.sh]: > [CASMPET-6582 14d9c5ad7] CASMPET-6582  Run shfmt
>  1 file changed, 471 insertions(+), 483 deletions(-)

is what happens when my editor autoformats. Way too many changes to put into a PR unless it's the only thing in the PR.

**Review by rustydb — APPROVED** [review]

**Pull request merged**